pkg/daemon/rpm-ostree.go

```diff
@@ -245,17 +297,82 @@ func (r *RpmOstreeClient) Rebase(imgURL, osImageContentDir string) (changed bool
 	args := []string{"rebase", "--experimental", fmt.Sprintf("%s:%s", repo, ostreeCsum),
 		"--custom-origin-url", customURL, "--custom-origin-description", "Managed by machine-config-operator"}
 
-	var out []byte
-	if out, err = runGetOut("rpm-ostree", args...); err != nil {
-		// capture stdout output as well in case of error
-		err = errors.Wrapf(err, "with stdout output: %v", string(out))
-		return
-	}
-
+	_, err = runRpmOstree(args[0], args[1:]...)
 	changed = true
 	return
 }
 
+// constant for describing kernel argument operations
+const (
+	kargRemove int = iota
+	kargAdd
+)
+
+// KernelArgument describes an operation for karg handling
+type KernelArgument struct {
+	Operation int
+	Name      string
+}
+
+// runRpmOStree wraps the rpm-ostree command. Unless mocked, this
+// is the rpmOstreeCommander that is used by RpmOstreeClient.
+func runRpmOstree(noun string, args ...string) ([]byte, error) {
+	if len(args) > 2 {
+		return nil, fmt.Errorf("rpm-ostree command is malformed, not enough arguments")
+	}
+	glog.Infof("Executing cmd: '%s %s'", realRpmOstreeCmd, args)
+	out, err := runGetOut(realRpmOstreeCmd, args...)
```

Comment:
I think because you don't pass the actual `noun` string here, any commands you pass to it will fail (since its not the full command), which in turn causes the very first invocation of `GetBootedDeployment` to fail since instead of
`rpm-ostree status --json` you give it `rpm-ostree --json`.

Reply:
I checked a daemon pod and sure enough it fails with

```F0221 21:54:09.287205   15334 start.go:150] Failed to initialize single run daemon: error reading osImageURL from rpm-ostree: error running /usr/bin/rpm-ostree --json```

I feel like since `noun` isn't referenced at all the golint should have caught it... weird

Reply:
Oof, yeah, I had that worked out but not pushed. The latest push should resolve this pain.